### ISSUE TITLE
A bug in Version matcher

### ISSUE THREAD

**Issue by jojostation75** [cloudfoundry/java-buildpack#872]
What expecting:
`JBP_CONFIG_AZURE_APPLICATION_INSIGHTS_AGENT: '{ version: 3.0.3 }'` should download exactly 3.0.3 version

What is now:
 `JBP_CONFIG_AZURE_APPLICATION_INSIGHTS_AGENT: '{ version: 3.0.3 }'` downloaded 3.0.3-BETA.3 version

**Note from other thread:**
That looks like a bug in the version matcher. I'd expect it to pick the exact match, but the version matcher is doing something like this:

1. Iterate through the list
2. Filter on whatever matches the pattern you've defined both 3.0.3 and 3.0.3_BETA.3 match your filter
3. Then it picks the max as defined by the Ruby max function, which ends up being 3.0.3_BETA.3

I was playing around with the unit tests to see if I could get it to match just 3.0.3 but I haven't been able to get it to match. The version doesn't allow a fixed start/end like with Regex ^ or $.

**Comment by JulianMei** [comment]
Hi @dmikusa-pivotal , any plans to fix this bug or any instructions on which files should be changed? thanks!

**Comment by dmikusa-pivotal** [comment]
Yes, it's on the roadmap but has been a lower priority issue since we haven't had a lot of people hit the issue. 

@JulianMei are you encountering this issue as well? If so, can you provide information similar to the original post? What env variable are you setting, a copy of your index.yml, what's the expected version to select, and what version actually gets selected. That'll help us with test cases. Thanks!
